libgstc/c/libgstc.c

```diff
@@ -983,3 +987,135 @@ gstc_pipeline_list (GstClient * client, char **pipelines[], int *list_lenght)
 out:
   return ret;
 }
+
+GstcStatus
+gstc_pipeline_list_signals (GstClient * client, const char *pipeline_name,
+    const char *element, char **signals[], int *list_lenght)
+{
+  GstcStatus ret;
+  char *what;
+  int asprintf_ret;
+  char *response;
+
+  gstc_assert_and_ret_val (NULL != client, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != pipeline_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != element, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != signals, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != list_lenght, GSTC_NULL_ARGUMENT);
+
+  asprintf_ret =
+      asprintf (&what, PIPELINE_SIGNAL_LIST_FORMAT, pipeline_name, element);
+  if (PRINTF_ERROR == asprintf_ret) {
+    return GSTC_OOM;
+  }
+
+  ret = gstc_cmd_read (client, what, &response, client->timeout);
+  if (GSTC_OK != ret) {
+    goto out;
+  }
+
+  ret = gstc_json_get_child_char_array (response, "response", "nodes",
+      "name", signals, list_lenght);
+
+out:
+  free (what);
+
+  return ret;
+}
+
+GstcStatus
+gstc_pipeline_signal_connect (GstClient * client, const char *pipeline_name,
+    const char *element, const char *signal, char **response)
+{
+  GstcStatus ret;
+  char *what;
+  int asprintf_ret;
+
+  gstc_assert_and_ret_val (NULL != client, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != pipeline_name, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != element, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != signal, GSTC_NULL_ARGUMENT);
+  gstc_assert_and_ret_val (NULL != response, GSTC_NULL_ARGUMENT);
+
+  asprintf_ret =
+      asprintf (&what, PIPELINE_SIGNAL_CONNECT_FORMAT, pipeline_name, element,
+      signal);
+  if (PRINTF_ERROR == asprintf_ret) {
+    return GSTC_OOM;
+  }
+
+  ret = gstc_cmd_read (client, what, response, client->timeout);
+
+  free (what);
+
+  return ret;
+}
+
+
+GstcStatus
+gstc_pipeline_signal_timeout (GstClient * client, const char *pipeline_name,
```

Comment:
Even if they are separate resources on the gstd resource tree, the pattern we have been following is that we absorb this configuration in a single API. For example, take a look at `GstcStatus gstc_pipeline_bus_wait (GstClient *client, const char *pipeline_name, const char *message_name, const long long timeout, char **message);`. It performs 3 gstd calls in a single API:
1. Configures the bus filter
2. Configures the bus timeout
3. Makes the bus read.

We should do something similar here.